internal/cluster/connection_manager.go

```diff
@@ -418,12 +419,13 @@ func (m *ConnectionManager) processAuthenticationResult(conn *Connection, result
 			return fmt.Sprintf("opened connection to: %s", connAddr)
 		})
 		m.eventDispatcher.Publish(NewConnectionOpened(conn))
+		return nil
 	case credentialsFailed:
-		return hzerrors.NewHazelcastAuthenticationError("invalid credentials", nil)
+		return fmt.Errorf("invalid credentials: %w", hzerrors.ErrAuthentication)
 	case serializationVersionMismatch:
-		return hzerrors.NewHazelcastAuthenticationError("serialization version mismatches with the server", nil)
+		return fmt.Errorf("serialization version mismatches with the server: %w", hzerrors.ErrAuthentication)
```

Comment:
Right, we do that in `checkAndSetPartitionCount` in `internal/cluster/partition_service.go`, around line 89.

Reply:
OK, good to know. What about the `NOT_ALLOWED_IN_CLUSTER` case? I remember that we have client blacklist feature in the Management Center and if a client is blacklisted, it might get this error.

The unique thing about this is that, if you got this error(`ClientNotAllowedInClusterException` in Java), you stop trying to connect the current cluster. I think we need to do the same in the Go client but maybe we can create an issue for it and implement it later

Reply:
Great bit of information, thanks. Let me think about that...

Reply:
I've added the not allowed in cluster case at: https://github.com/hazelcast/hazelcast-go-client/pull/586/commits/b51ca4c6c8f8229f07f7fe45466060d58350661b